Helper/MerchantMetrics.php

```diff
@@ -0,0 +1,343 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Helper;
+
+use http\Encoding\Stream;
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Bolt\Boltpay\Helper\Config as ConfigHelper;
+use Magento\Framework\Filesystem\DirectoryList;
+use Magento\Store\Model\StoreManagerInterface;
+use Bolt\Boltpay\Helper\Log as LogHelper;
+
+
+
+/**
+ * Boltpay Merchant Metrics wrapper helper
+ *
+ * @SuppressWarnings(PHPMD.TooManyFields)
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+
+class MerchantMetrics extends AbstractHelper
+{
+    const STAGE_DEVELOPMENT = 'development';
+    const STAGE_PRODUCTION  = 'production';
+
+    /**
+     * @var \GuzzleHttp\Client
+     */
+    private $guzzleClient;
+
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var string
+     */
+    private $metricsFile;
+
+    /**
+     * @var LogHelper
+     */
+    private $logHelper;
+
+     /**
+     * @var array
+     */
+    public $metrics;
+
+     /**
+     * @var array
+     */
+    private $headers;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @param Context $context
+     * @param Config $configHelper
+     * @param DirectoryList $directoryList
+     * @param StoreManagerInterface $storeManager
+     * @param Bugsnag $bugsnag
+     * @param LogHelper $logHelper
+     *
+     *
+     * @throws
+     */
+    public function __construct(
+        Context $context,
+        ConfigHelper $configHelper,
+        DirectoryList $directoryList,
+        StoreManagerInterface $storeManager,
+        Bugsnag $bugsnag,
+        LogHelper $logHelper
+    ) {
+        parent::__construct($context);
+
+        $this->storeManager = $storeManager;
+        $this->bugsnag = $bugsnag;
+        $this->logHelper = $logHelper;
+        //////////////////////////////////////////
+        // Composerless installation.
+        // Make sure libraries are in place:
+        // lib/internal/Bolt/bugsnag
+        // lib/internal/Bolt/guzzle
+        //////////////////////////////////////////
+        if (!class_exists('\GuzzleHttp\Client')) {
+            // phpcs:ignore
+            require_once $directoryList->getPath('lib_internal') . '/Bolt/guzzle/autoloader.php';
+        }
+        $this->configHelper = $configHelper;
+        $this->metrics = array();
+        $this->metricsFile = null;
+        $this->guzzleClient = null;
+        $this->metricsFile = null;
```

Comment:
why not initialize it here?

Reply:
Again here is for mocking, any suggestions to avoid?